src/test/java/org/mockitousage/stubbing/StrictStubbingEndToEndTest.java

```diff
@@ -42,6 +42,12 @@
                 .fails("unnecessary_stubbing", UnnecessaryStubbingException.class);
     }
 
+    @Test public void unnecessary_stubbing_with_eq_implicit_matchers() {
```

Comment:
This is a good e2e test.

Can you add coverage for the same use case but with the "willReturn(x).when(...)" stubbing style? This should "just" work without new changes but adding extra coverage would be nice for our codebase :-)

Reply:
Done.